### DISCUSSION TITLE
Receiving reactive objects in the repository interface

### DISCUSSION THREAD

**Discussion by averri** [micronaut-projects/micronaut-data#2045] [Q&A] [answered]
Is it possible to save/update using reactive objects in the function arguments of reactive repositories? 

Example: I would like to be able to save an entity of class `com.mydomain.entity.User` using:

```kotlin
import com.mydomain.entity.User
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.r2dbc.annotation.R2dbcRepository
import io.micronaut.data.repository.reactive.ReactiveStreamsPageableRepository
import org.reactivestreams.Publisher

@R2dbcRepository(dialect = Dialect.POSTGRES)
interface UserRepository: ReactiveStreamsPageableRepository<Long, User> {
    
    fun save(user: Publisher<User>): Publisher<User>
}
```

This is extremely useful for use in conjunction with reactive controllers, which can accept function arguments using `Publisher<T>` where `T` is the generic object type.

**Comment by dstepanov** [comment, marked as answer]
It's not possible right now. Please create an issue.

**Reply by averri** [reply]
Thanks, @dstepanov. I have created this one, let me know if the information is not clear:
https://github.com/micronaut-projects/micronaut-data/issues/2049
